File: services/hpc-service/src/hpc/auth/AuthClient.java
package hpc.auth;

import java.io.IOException;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class AuthClient {

    private final String baseUrl;

    private final HttpClient client;

    public AuthClient(
        String baseUrl
    ) {

        this.baseUrl =
            baseUrl.replaceAll(
                "/+$",
                ""
            );

        this.client =
            HttpClient
                .newBuilder()
                .connectTimeout(
                    Duration.ofSeconds(
                        3
                    )
                )
                .build();
    }

    public AuthIdentity validate(
        String token
    ) throws IOException,
        InterruptedException {

        if (
            token == null ||
            token.isBlank()
        ) {

            return new AuthIdentity(
                false,
                "",
                "",
                "Token requerido"
            );
        }

        URI uri =
            URI.create(
                baseUrl +
                "/internal/auth/validate"
            );

        HttpRequest request =
            HttpRequest
                .newBuilder(
                    uri
                )
                .header(
                    "Authorization",
                    "Bearer " + token
                )
                .GET()
                .timeout(
                    Duration.ofSeconds(
                        5
                    )
                )
                .build();

        HttpResponse<String> response =
            client.send(
                request,
                HttpResponse
                    .BodyHandlers
                    .ofString()
            );

        if (
            response.statusCode()
                != 200
        ) {

            throw new IOException(
                "Auth Service respondió HTTP " +
                response.statusCode()
            );
        }

        String[] parts =
            response
                .body()
                .trim()
                .split(
                    "\\|",
                    4
                );

        if (
            parts.length < 3
        ) {

            throw new IOException(
                "Respuesta inválida de Auth Service"
            );
        }

        boolean valid =
            Boolean.parseBoolean(
                parts[0]
            );

        String userId =
            parts.length > 1
                ? parts[1]
                : "";

        String role =
            parts.length > 2
                ? parts[2]
                : "";

        String message =
            parts.length > 3
                ? parts[3]
                : "";

        return new AuthIdentity(
            valid,
            userId,
            role,
            message
        );
    }
}

File: services/stream-service/src/AuthClient.php
<?php

declare(strict_types=1);

final class AuthClient
{
    private string $baseUrl;

    public function __construct(
        string $baseUrl
    ) {
        $this->baseUrl =
            rtrim(
                $baseUrl,
                '/'
            );
    }

    // =====================================
    // VALIDATE TOKEN
    // =====================================

    public function validateToken(
        string $token
    ): array {
        $token =
            trim(
                $token
            );

        if ($token === '') {
            throw new RuntimeException(
                'Token obligatorio'
            );
        }

        $url =
            $this->baseUrl .
            '/internal/auth/validate';

        $context =
            stream_context_create([
                'http' => [
                    'method' =>
                        'GET',

                    'timeout' =>
                        3,

                    'ignore_errors' =>
                        true,

                    'header' =>
                        "Authorization: Bearer " .
                        $token . "\r\n",
                ],
            ]);

        $body =
            @file_get_contents(
                $url,
                false,
                $context
            );

        if ($body === false) {
            throw new RuntimeException(
                'No fue posible conectar con Auth Service'
            );
        }

        $parts =
            explode(
                '|',
                trim($body),
                4
            );

        if (count($parts) !== 4) {
            throw new RuntimeException(
                'Respuesta inválida de Auth Service'
            );
        }

        return [
            'valid' =>
                strtolower(
                    trim($parts[0])
                ) === 'true',

            'userId' =>
                trim(
                    $parts[1]
                ),

            'role' =>
                strtoupper(
                    trim(
                        $parts[2]
                    )
                ),

            'message' =>
                trim(
                    $parts[3]
                ),
        ];
    }
}
